### PR TITLE
Add status checks for the database and sqs services to the /livez endpoint

### PR DESCRIPTION
**Pull request by palasanu**
## Description

Add status checks for the database and sqs services to the /livez endpoint.
With it, I also added two metrics that count the errors from the status checks: database_status_error_count, sqs_status_error_count

## Motivation and Context

Before the endpoint just checked if the API was up and running, now there are connectivity checks that can identify if one of the underlying services can be reached.

## How Has This Been Tested?

Tested the changes locally with mocks.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by palasanu** [comment]
Rerun tests

File: pkg/api/database/database_test.go [hunk 1]
```diff
@@ -106,6 +111,36 @@ func TestNewDb(t *testing.T) {
...
+	tcs := []struct {
+		name          string
+		tableName     string
+		expectedError error
+	}{
+		{
+			name:          "unhealthy hatabase test",
+			tableName:     "mock-clusters",
+			expectedError: errors.New("No sqs found with the name mock-clusters"),
+		},
```
Comment (victorvarza) [inline pkg/api/database/database_test.go:127]: Let's add also a happy path test.

File: pkg/api/monitoring/metrics.go [hunk 2]
```diff
@@ -71,6 +71,27 @@ var egressMetrics = []*Metric{
 	egressReqDur,
 }
 
+var databaseStatusErrCnt = &Metric{
+	ID:          "DatabaseStatusErrCnt",
```
Comment (victorvarza) [inline pkg/api/monitoring/metrics.go:75]: What is the benefit of recording this metric?

File: pkg/api/monitoring/metrics.go [hunk 3]
```diff
@@ -71,6 +71,27 @@ var egressMetrics = []*Metric{
...
+	Name:        "database_status_error_count",
+	Description: "The total number of errors returned from the database for every /livez check",
+	Type:        "counter_vec",
+	Args:        []string{"target"},
+}
+
+var sqsStatusErrCnt = &Metric{
+	ID:          "SqsStatusErrCnt",
```
Comment (victorvarza) [inline pkg/api/monitoring/metrics.go:83]: Same here.

File: pkg/api/monitoring/metrics.go [hunk 4]
```diff
@@ -110,13 +131,16 @@ type MetricsI interface {
...
 	Use(e *echo.Echo)
 }
 
 // Metrics contains the metrics gathered by the instance and its path
 type Metrics struct {
-	ingressReqCnt, egressReqCnt *prometheus.CounterVec
-	ingressReqDur, egressReqDur *prometheus.HistogramVec
+	ingressReqCnt, egressReqCnt           *prometheus.CounterVec
+	ingressReqDur, egressReqDur           *prometheus.HistogramVec
+	databaseStatusErrCnt, sqsStatusErrCnt *prometheus.CounterVec
```
Comment (victorvarza) [inline pkg/api/monitoring/metrics.go:143]: We should't have specifics in this module. Instead we can record errors in general as `errCnt`

**Review by victorvarza — CHANGES_REQUESTED** [review]
Looks good overall! Let't discuss the metric for target errors.

**Review by victorvarza — APPROVED** [review]

**Pull request merged**